### PR TITLE
add test case for embeded constructors; re #1101

### PR DESCRIPTION
**Pull request by compleatang**
Signed-off-by: Casey Kuhlman <casey@monax.io>

### CODE REVIEW

File: tests/jobs_fixtures/app56-embedded-constructors/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+- name: deployD
+  deploy:
+    contract: contract.sol
+    instance: D
+
+- name: deployX
+  deploy:
+    contract: contract.sol
+    instance: X
```
Comment (seanyoung) [inline tests/jobs_fixtures/app56-embedded-constructors/deploy.yaml]: The steps deployD and deployX are not necesary. When you deploy a contract that creates other contracts, those other contract need not be deployed at all -- they're copied verbatim into the contract bytecode.

Note these two steps are interesting for a completely different reason. The deploy of these contracts has no arguments for the constructor, but it does not fail. burrow deploy should warn about this, now the address argument is 0 (presumably).

**Comment by fastchain** [comment]
Hi,
I'm trying to run this test on burrow 0.25.0+commit.v0.24.3-78-gc44fbec9+2019-05-0 

with this script  https://chat.hyperledger.org/channel/burrow?msg=asJMpTJmvWSKGEFXC

and I get this 

```
[afx@proton bin]$ ./dep.sh 
20703A7FDA5C5566EC487D05E50FBB15436BD6EF
log_channel=Info message="Using chain" Chain=127.0.0.1:10997 Signer=
log_channel=Info message="Loading Playbook File."
log_channel=Info message="Loading playbook file" path=/home/afx/go/src/github.com/hyperledger/burrow/bin filename=/home/afx/go/src/github.com/hyperledger/burrow/bin/deploy.yaml
log_channel=Info message="*****Executing Job*****" JobName=defaultAddr Type=Account
log_channel=Info message="Setting Account" account=20703A7FDA5C5566EC487D05E50FBB15436BD6EF
log_channel=Info message="*****Executing Job*****" JobName=deployD Type=Deploy
ERROR: Could not find contract in any of [contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/bin/contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/bin/contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/contract.sol]log_channel=Info message="JOBS THAT FAILED" count=1
log_channel=Info message="Playbook result" jobNo=0 file=deploy.yaml error="Could not find contract in any of [contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/bin/contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/bin/contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/contract.sol /home/afx/go/src/github.com/hyperledger/burrow/bin/contract.sol]" time=1.345762ms
```

borrow.log:
```
Sourcing config from first of: Config file at 'config.toml' then defaults
Sourcing config from Config file at 'config.toml'
Using validator address: 20703A7FDA5C5566EC487D05E50FBB15436BD6EF
{"caller":"blockchain.go:75","database_key":"426C6F636B636861696E5374617465","height":0,"log_channel":"Info","message":"Trying to load blockchain state from database","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"LoadOrNewBlockchain","time":"2019-05-01T06:09:14.060234723Z"}
{"caller":"blockchain.go:92","height":0,"log_channel":"Info","message":"No existing blockchain state found in database, making new blockchain","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"LoadOrNewBlockchain","time":"2019-05-01T06:09:14.060665492Z"}
{"caller":"kernel.go:134","height":0,"log_channel":"Info","message":"Creating new application state from genesis","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewKernel()","time":"2019-05-01T06:09:14.062143963Z"}
{"caller":"kernel.go:145","height":0,"log_channel":"Info","message":"State loading successful","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewKernel()","time":"2019-05-01T06:09:14.063749216Z"}
{"caller":"app.go:133","component":"ABCI_App","height":0,"log_channel":"Info","message":"Initial validator set matches","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-61FECA_ValidatorID:20703A7FDA5C5566EC487D05E50FBB15436BD6EF","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"abci.NewApp","time":"2019-05-01T06:09:14.257245838Z"}
{"announce":"startup","burrow_version":"0.25.0+commit.v0.24.3-78-gc44fbec9+2019-05-01","caller":"processes.go:195","genesis_app_hash":"","genesis_hash":"61FECA34443953643AB4AD32DA0D05BDEC515D3F4EDD7C462957CB7961CD15F6","height":0,"launch_time":"2019-05-01 15:09:14.350281949 +0900 JST m=+0.382867141","log_channel":"Info","message":"Burrow is launching. We have marmot-off.","net_address":"658ea986d1cd15bfd6b00c13b8fbf5c8f2ead17c@0.0.0.0:26656","node_id":"658ea986d1cd15bfd6b00c13b8fbf5c8f2ead17c","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewKernel()","tendermint_version":"0.31.5","time":"2019-05-01T06:09:14.350914237Z","validator_address":"20703A7FDA5C5566EC487D05E50FBB15436BD6EF"}
{"caller":"http_server.go:19","component":"RPC_Info","height":0,"listen_address":"127.0.0.1:26658","log_channel":"Info","message":"Starting RPC HTTP server","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewKernel()","time":"2019-05-01T06:09:14.351779148Z"}
{"block_time":"2019-05-01 06:09:13 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"8C105979F9B00667B779ECABE8A2D295D9CFDC81F3FD47EE77631D657BD2C2F9","height":1,"last_block_duration":"0s","last_block_hash":"","last_block_time":"2019-05-01 06:09:13 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-61FECA_ValidatorID:20703A7FDA5C5566EC487D05E50FBB15436BD6EF","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Commit()","tag":"Commit","time":"2019-05-01T06:09:14.795857905Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":1,"log_channel":"Info","message":"Executor committing","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewBatchCommitter","time":"2019-05-01T06:09:14.796231923Z"}
{"caller":"app.go:313","component":"ABCI_App","height":1,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-61FECA_ValidatorID:20703A7FDA5C5566EC487D05E50FBB15436BD6EF","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"abci.NewApp","time":"2019-05-01T06:09:14.814622556Z"}
{"caller":"kernel.go:332","height":1,"log_channel":"Info","message":"Caught terminated signal so shutting down","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewKernel()","signal":"terminated","time":"2019-05-01T06:09:15.063320993Z"}
{"caller":"kernel.go:355","height":1,"log_channel":"Info","message":"Attempting graceful shutdown...","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","time":"2019-05-01T06:09:15.063434056Z"}
{"caller":"kernel.go:356","height":1,"log_channel":"Info","message":"Shutting down servers","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","time":"2019-05-01T06:09:15.063522425Z"}
{"caller":"kernel.go:362","height":1,"log_channel":"Info","message":"Shutting down server","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","server_name":"rpcConfig/GRPC","time":"2019-05-01T06:09:15.063568164Z"}
{"caller":"kernel.go:362","height":1,"log_channel":"Info","message":"Shutting down server","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","server_name":"rpcConfig/info","time":"2019-05-01T06:09:15.063736924Z"}
{"caller":"kernel.go:362","height":1,"log_channel":"Info","message":"Shutting down server","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","server_name":"StartupAnnouncer","time":"2019-05-01T06:09:15.063815656Z"}
{"announce":"shutdown","caller":"processes.go:163","elapsed_run_time":"713.569627ms","height":1,"log_channel":"Info","message":"Burrow is shutting down. Prepare for re-entrancy.","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewKernel()","shutdown_time":"2019-05-01 15:09:15.063851684 +0900 JST m=+1.096436768","time":"2019-05-01T06:09:15.063854408Z"}
{"caller":"kernel.go:362","height":1,"log_channel":"Info","message":"Shutting down server","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","server_name":"Tendermint","time":"2019-05-01T06:09:15.063894951Z"}
{"caller":"processes.go:147","height":1,"log_channel":"Info","message":"Tendermint Node has quit, closing DB connections...","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"NewKernel()","time":"2019-05-01T06:09:15.233886022Z"}
{"caller":"kernel.go:362","height":1,"log_channel":"Info","message":"Shutting down server","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","server_name":"Database","time":"2019-05-01T06:09:15.235217235Z"}
{"caller":"kernel.go:374","height":1,"log_channel":"Info","message":"Shutdown complete","run_id":"a4e42bbd-6bd7-11e9-8505-d312d2f1ae3a","scope":"Shutdown","time":"2019-05-01T06:09:15.235591906Z"}
```


What am I doing wrong ? 
Thanks

**Comment by seanyoung** [comment]
@fastchain you want the contract.sol in the same location as the deploy.yaml

**Comment by fastchain** [comment]
@seanyoung  thanks, I've missed it.  But now there is another one:

```
afx@proton ~/g/s/g/h/b/bin> ./dep.sh
C6CE6DA06446897CD6F0F7BD781DF56169C09BDF
log_channel=Info message="Using chain" Chain=127.0.0.1:10997 Signer=
log_channel=Info message="Loading Playbook File."
log_channel=Info message="Loading playbook file" path=/home/afx/go/src/github.com/hyperledger/burrow/bin filename=/home/afx/go/src/github.com/hyperledger/burrow/bin/deploy.yaml
log_channel=Info message="*****Executing Job*****" JobName=defaultAddr Type=Account
log_channel=Info message="Setting Account" account=C6CE6DA06446897CD6F0F7BD781DF56169C09BDF
log_channel=Info message="*****Executing Job*****" JobName=deployD Type=Deploy
log_channel=Info message="Contract path" path=contract.sol
ERROR: exit status 1log_channel=Info message="JOBS THAT FAILED" count=1
log_channel=Info message="Playbook result" jobNo=0 file=deploy.yaml error="exit status 1" time=1.986298134s

```

borrow.log:
```
Sourcing config from first of: Config file at 'config.toml' then defaults
Sourcing config from Config file at 'config.toml'
Using validator address: C6CE6DA06446897CD6F0F7BD781DF56169C09BDF
{"caller":"blockchain.go:75","database_key":"426C6F636B636861696E5374617465","height":0,"log_channel":"Info","message":"Trying to load blockchain state from database","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"LoadOrNewBlockchain","time":"2019-05-01T08:29:32.112181626Z"}
{"caller":"blockchain.go:92","height":0,"log_channel":"Info","message":"No existing blockchain state found in database, making new blockchain","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"LoadOrNewBlockchain","time":"2019-05-01T08:29:32.112520799Z"}
{"caller":"kernel.go:134","height":0,"log_channel":"Info","message":"Creating new application state from genesis","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewKernel()","time":"2019-05-01T08:29:32.113233283Z"}
{"caller":"kernel.go:145","height":0,"log_channel":"Info","message":"State loading successful","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewKernel()","time":"2019-05-01T08:29:32.114344544Z"}
{"caller":"app.go:133","component":"ABCI_App","height":0,"log_channel":"Info","message":"Initial validator set matches","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:32.279504984Z"}
{"announce":"startup","burrow_version":"0.25.0+commit.v0.24.3-78-gc44fbec9+2019-05-01","caller":"processes.go:195","genesis_app_hash":"","genesis_hash":"2926F50CA650CCCF542252F3A97128AABE0C074E96976ECFEC80900EEF2CD1BE","height":0,"launch_time":"2019-05-01 17:29:32.359327816 +0900 JST m=+0.324918630","log_channel":"Info","message":"Burrow is launching. We have marmot-off.","net_address":"253e5199588212b2634c99bb5b43513682d0b2df@0.0.0.0:26656","node_id":"253e5199588212b2634c99bb5b43513682d0b2df","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewKernel()","tendermint_version":"0.31.5","time":"2019-05-01T08:29:32.359826513Z","validator_address":"C6CE6DA06446897CD6F0F7BD781DF56169C09BDF"}
{"caller":"http_server.go:19","component":"RPC_Info","height":0,"listen_address":"127.0.0.1:26658","log_channel":"Info","message":"Starting RPC HTTP server","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewKernel()","time":"2019-05-01T08:29:32.36042474Z"}
{"block_time":"2019-05-01 08:29:32 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"E4D134EBEAD975E6A14E9D5B51407738EF0BCAD788CFC2D820B67D5CA9DA92EF","height":1,"last_block_duration":"0s","last_block_hash":"","last_block_time":"2019-05-01 08:29:32 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Commit()","tag":"Commit","time":"2019-05-01T08:29:32.789478035Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":1,"log_channel":"Info","message":"Executor committing","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewBatchCommitter","time":"2019-05-01T08:29:32.789985934Z"}
{"caller":"app.go:313","component":"ABCI_App","height":1,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:32.805525356Z"}
{"block_time":"2019-05-01 08:29:32.733394167 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"C426B432279E7E90A0A4CAB8A9207E592BC73C72FBD4746D5E09DEF6DAD1BEBA","height":2,"last_block_duration":"0s","last_block_hash":"E4D134EBEAD975E6A14E9D5B51407738EF0BCAD788CFC2D820B67D5CA9DA92EF","last_block_time":"2019-05-01 08:29:32 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Commit()","tag":"Commit","time":"2019-05-01T08:29:33.179218832Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":2,"log_channel":"Info","message":"Executor committing","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewBatchCommitter","time":"2019-05-01T08:29:33.179339502Z"}
{"caller":"app.go:313","component":"ABCI_App","height":2,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:33.190023651Z"}
{"block_time":"2019-05-01 08:29:33.133907513 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"2806673DD1B0BE4CE24655FE722D674D14AE47316E8C93865A4347CBA02801DA","height":3,"last_block_duration":"733.394167ms","last_block_hash":"C426B432279E7E90A0A4CAB8A9207E592BC73C72FBD4746D5E09DEF6DAD1BEBA","last_block_time":"2019-05-01 08:29:32.733394167 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Commit()","tag":"Commit","time":"2019-05-01T08:29:33.563950313Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":3,"log_channel":"Info","message":"Executor committing","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewBatchCommitter","time":"2019-05-01T08:29:33.56405729Z"}
{"caller":"app.go:313","component":"ABCI_App","height":3,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:33.571310542Z"}
{"block_time":"2019-05-01 08:29:33.523210933 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"11657D90548B443D957ECBE64454EA3E74167402E9F2506AB10CEE67C834BE7A","height":4,"last_block_duration":"400.513346ms","last_block_hash":"2806673DD1B0BE4CE24655FE722D674D14AE47316E8C93865A4347CBA02801DA","last_block_time":"2019-05-01 08:29:33.133907513 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Commit()","tag":"Commit","time":"2019-05-01T08:29:33.950586061Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":4,"log_channel":"Info","message":"Executor committing","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewBatchCommitter","time":"2019-05-01T08:29:33.950695709Z"}
{"caller":"app.go:313","component":"ABCI_App","height":4,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:33.961223783Z"}
{"block_time":"2019-05-01 08:29:33.905850465 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"EB3FC148AA0A936BFE1077CA3C5234521613A729B50D5B526790E2DDA9EE3B1E","height":5,"last_block_duration":"389.30342ms","last_block_hash":"11657D90548B443D957ECBE64454EA3E74167402E9F2506AB10CEE67C834BE7A","last_block_time":"2019-05-01 08:29:33.523210933 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Commit()","tag":"Commit","time":"2019-05-01T08:29:34.326229671Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":5,"log_channel":"Info","message":"Executor committing","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewBatchCommitter","time":"2019-05-01T08:29:34.32632517Z"}
{"caller":"app.go:313","component":"ABCI_App","height":5,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:34.335048925Z"}
{"block_time":"2019-05-01 08:29:34.286307218 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"0EADD1382FC4318B7813BC661C24CEB969A92DD63C3901DFBE03A719B85E2E15","height":6,"last_block_duration":"382.639532ms","last_block_hash":"EB3FC148AA0A936BFE1077CA3C5234521613A729B50D5B526790E2DDA9EE3B1E","last_block_time":"2019-05-01 08:29:33.905850465 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Commit()","tag":"Commit","time":"2019-05-01T08:29:34.721593993Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":6,"log_channel":"Info","message":"Executor committing","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewBatchCommitter","time":"2019-05-01T08:29:34.721813067Z"}
{"caller":"app.go:313","component":"ABCI_App","height":6,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:34.733788259Z"}
{"block_time":"2019-05-01 08:29:34.671991258 +0000 UTC","caller":"app.go:262","component":"ABCI_App","hash":"073841B994790EAD7C4E791904C7A7509E36E42756BD5D15D07B3DF41599CAEC","height":7,"last_block_duration":"380.456753ms","last_block_hash":"0EADD1382FC4318B7813BC661C24CEB969A92DD63C3901DFBE03A719B85E2E15","last_block_time":"2019-05-01 08:29:34.286307218 +0000 UTC","log_channel":"Info","message":"Committing block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Commit()","tag":"Commit","time":"2019-05-01T08:29:35.097079287Z","txs":0}
{"caller":"execution.go:311","component":"Executor","height":7,"log_channel":"Info","message":"Executor committing","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewBatchCommitter","time":"2019-05-01T08:29:35.097248283Z"}
{"caller":"app.go:313","component":"ABCI_App","height":7,"log_channel":"Info","message":"Committed block","node_info":"Burrow_0.25.0_BurrowChain_B6FC16-2926F5_ValidatorID:C6CE6DA06446897CD6F0F7BD781DF56169C09BDF","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"abci.NewApp","time":"2019-05-01T08:29:35.106748876Z"}
{"caller":"kernel.go:332","height":7,"log_channel":"Info","message":"Caught terminated signal so shutting down","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewKernel()","signal":"terminated","time":"2019-05-01T08:29:35.104264995Z"}
{"caller":"kernel.go:355","height":7,"log_channel":"Info","message":"Attempting graceful shutdown...","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","time":"2019-05-01T08:29:35.110800219Z"}
{"caller":"kernel.go:356","height":7,"log_channel":"Info","message":"Shutting down servers","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","time":"2019-05-01T08:29:35.110872536Z"}
{"caller":"kernel.go:362","height":7,"log_channel":"Info","message":"Shutting down server","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","server_name":"rpcConfig/GRPC","time":"2019-05-01T08:29:35.110889136Z"}
{"caller":"kernel.go:362","height":7,"log_channel":"Info","message":"Shutting down server","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","server_name":"rpcConfig/info","time":"2019-05-01T08:29:35.110953348Z"}
{"caller":"kernel.go:362","height":7,"log_channel":"Info","message":"Shutting down server","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","server_name":"StartupAnnouncer","time":"2019-05-01T08:29:35.111003576Z"}
{"announce":"shutdown","caller":"processes.go:163","elapsed_run_time":"2.751700433s","height":7,"log_channel":"Info","message":"Burrow is shutting down. Prepare for re-entrancy.","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewKernel()","shutdown_time":"2019-05-01 17:29:35.111028326 +0900 JST m=+3.076619063","time":"2019-05-01T08:29:35.111029829Z"}
{"caller":"kernel.go:362","height":7,"log_channel":"Info","message":"Shutting down server","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","server_name":"Tendermint","time":"2019-05-01T08:29:35.111066309Z"}
{"caller":"processes.go:147","height":7,"log_channel":"Info","message":"Tendermint Node has quit, closing DB connections...","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"NewKernel()","time":"2019-05-01T08:29:35.130690497Z"}
{"caller":"kernel.go:362","height":7,"log_channel":"Info","message":"Shutting down server","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","server_name":"Database","time":"2019-05-01T08:29:35.131007131Z"}
{"caller":"kernel.go:374","height":7,"log_channel":"Info","message":"Shutdown complete","run_id":"3e72848e-6beb-11e9-8047-44b00848496f","scope":"Shutdown","time":"2019-05-01T08:29:35.131152279Z"}
```

**Comment by seanyoung** [comment]
First of all can we use rocketchat please, a closed pull request isn't the most appopriate.

Looks like the error message needs improving but I think solc is not in your path or is not working. What does `solc --version` say?

**Comment by fastchain** [comment]
>First of all can we use rocketchat please, a closed pull request isn't the most appopriate.

Sure!  what is rocket chat?

>solc --version say?

0.4.24+commit.e67f0147.Emscripten.clang 

So it seems  the problem is here. 
Is it possible somehow to use compiled contract (abi,bin) in this setup? (I would like to use online compiler to make reproducible tests)

**Comment by seanyoung** [comment]
@fastchain https://chat.hyperledger.org/channel/burrow
You need the native solc binary, not the emscripten version. 

If you type `make bin/solc` in the burrow source directory, it will download it. Then copy `bin/solc` into your path somewhere. If you have run a build of burrow it should already be there.

**Pull request merged**